includes/calendars/views/default-calendar-grid.php

```diff
@@ -363,22 +363,53 @@ private function draw_month($month, $year, $id = 0)
 			$timestamps = array_keys($events);
 			$lower_bound = array_filter($timestamps, [$this, 'filter_events_before']);
 			$higher_bound = array_filter($lower_bound, [$this, 'filter_events_after']);
-			$filtered =
-				is_array($events) && is_array($higher_bound) && !empty($events) && !empty($higher_bound)
-					? array_intersect_key($events, array_combine($higher_bound, $higher_bound))
-					: [];
 
-			// Put resulting events in an associative array, with day of the month as key for easy retrieval in calendar days loop.
+			$current_month = Carbon::createFromTimestamp($this->start, $calendar->timezone)->month;
```

Comment:
I don't understand the changes starting from this line till 382. Can you please explain?